mvcc/backend/read_tx.go

```diff
@@ -40,10 +40,12 @@ type readTx struct {
 	mu  sync.RWMutex
 	buf txReadBuffer
 
-	// txmu protects accesses to buckets and tx on Range requests.
-	txmu    sync.RWMutex
+	// txMu protects accesses to buckets and tx on Range requests.
```

Comment:
+1 to the general idea from the "create a base type for readTx and concurrentReadTx to avoid duplicated function implementation" TODO. In the meantime, copy these field comments to concurrentReadTx?

Maybe what we need to do is encapsulate the `bolt.Tx` and it's corresponding buckets and mutex into a struct so we don't need to have three separate fields that are not entirely obviously related?

Reply:
+1 to encapsulation. I had the same idea when developing this PR.

Reply:
Added TODO, will address in future PRs.